appservice/src/deploy/runWithZipStream.ts

```diff
@@ -34,8 +34,15 @@ export async function runWithZipStream(context: IActionContext, fsPath: string,
         });
     } else {
         ext.outputChannel.appendLog(localize('zipCreate', 'Creating zip package...'), { resourceName: client.fullName });
-        const zipFile = new yazl.ZipFile();
+        const zipFile: yazl.ZipFile & { outputStreamCursor?: number } = new yazl.ZipFile();
         let filesToZip: string[] = [];
+        let sizeOfZipFile: number = 0;
+
+        zipFile.outputStream.on('data', (chunk) => {
+            sizeOfZipFile += chunk.length
```

Comment:
With my test folder, I was seeing it zipped by 8.19 MB vs. 7.82 MB archived by winrar at normal with zip compression.